packages/shared/src/hooks/analytics/useAnalyticsSharedProps.ts

```diff
@@ -28,6 +28,7 @@ const getOrGenerateDeviceId = async (): Promise<string> => {
 export default function useAnalyticsSharedProps(
   app: string,
   version: string,
+  deviceId?: string,
```

Comment:
It is better to not mix two ways of using this hook:
* Getting the device id as an argument
* The hook loads the device id from storage by itself
Due to the companion constraints, we have to use the first method so I suggest applying it throughout the application

Reply:
I'm a bit stuck what's wisdom here.
If we use option1 how will we pass it?

As the current `getOrGenerateDeviceId` is a promise that returns a string.
I don't want to pre-block from the parent components, nor have the option the deviceId is not set yet.

Perhaps passing a promise as the `deviceId` option could be an option?
That way we can either pass the initial one, or a plain string resolve?

Reply:
I actually created a new look for the device ID, let me know how that looks.